sds/src/scanner/mod.rs

```diff
@@ -523,15 +528,26 @@ impl Scanner {
     ) -> Result<Vec<RuleMatch>, ScannerError> {
         let start = Instant::now();
         let result = self.internal_scan(event, options).await;
-        match &result {
-            Ok(rule_matches) => {
-                self.record_metrics(rule_matches, start);
+        match result {
+            Ok((rule_matches, io_duration)) => {
+                self.record_metrics(&rule_matches, start);
+                let total_duration = start.elapsed();
+
+                // Calculate CPU duration by subtracting I/O wait time from total duration
+                let cpu_duration = total_duration.saturating_sub(io_duration);
+
+                // Record CPU duration histogram in nanoseconds
+                self.metrics
+                    .cpu_duration
+                    .record(cpu_duration.as_nanos() as f64);
```

Comment:
Nit: self explanatory code, not need to add comments on what the code is doing, comments should be needed only to explain why such code is needed 
```suggestion
                self.metrics
                    .cpu_duration
                    .record(cpu_duration.as_nanos() as f64);
```